synapse/server_notices/server_notices_manager.py

```diff
@@ -119,7 +119,7 @@ async def get_or_create_notice_room_for_user(self, user_id: str) -> str:
             # manages to invite the system user to a room, that doesn't make it
             # the server notices room.
             user_ids = await self._store.get_users_in_room(room.room_id)
-            if self.server_notices_mxid in user_ids:
+            if len(user_ids) == 2 and self.server_notices_mxid in user_ids:
```

Comment:
```suggestion
            if len(user_ids) <= 2 and self.server_notices_mxid in user_ids:
```

As the user may not have joined yet.